testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/servlet/UserStorageConsentTest.java

```diff
@@ -108,26 +108,27 @@ public static void setupConsent(KeycloakSession session) {
         product.setConsentRequired(true);
         ClientScopeModel clientScope = realm.addClientScope("clientScope");
         clientScope.setProtocol(OIDCLoginProtocol.LOGIN_PROTOCOL);
-        System.err.println("client scope protocol mappers size: " + clientScope.getProtocolMappers().size());
-
-        for (ProtocolMapperModel mapper : product.getProtocolMappers()) {
-            if (mapper.getProtocol().equals(OIDCLoginProtocol.LOGIN_PROTOCOL)) {
-                if (mapper.getName().equals(OIDCLoginProtocolFactory.USERNAME)
-                        || mapper.getName().equals(OIDCLoginProtocolFactory.EMAIL)
-                        || mapper.getName().equals(OIDCLoginProtocolFactory.GIVEN_NAME)
-                        ) {
+        System.err.println("client scope protocol mappers size: " + clientScope.getProtocolMappersStream().count());
+
+        product.getProtocolMappersStream()
+                .filter(mapper -> Objects.equals(OIDCLoginProtocol.LOGIN_PROTOCOL, mapper.getProtocol()))
+                .filter(mapper -> Objects.equals(OIDCLoginProtocolFactory.USERNAME, mapper.getName()) ||
+                        Objects.equals(OIDCLoginProtocolFactory.EMAIL, mapper.getName()) ||
+                        Objects.equals(OIDCLoginProtocolFactory.GIVEN_NAME, mapper.getName()))
+                .map(mapper -> {
                     ProtocolMapperModel copy = new ProtocolMapperModel();
                     copy.setName(mapper.getName());
                     copy.setProtocol(mapper.getProtocol());
                     Map<String, String> config = new HashMap<>();
                     config.putAll(mapper.getConfig());
                     copy.setConfig(config);
                     copy.setProtocolMapper(mapper.getProtocolMapper());
-                    clientScope.addProtocolMapper(copy);
-                }
-            }
-            product.removeProtocolMapper(mapper);
-        }
+
+                    product.removeProtocolMapper(mapper);
```

Comment:
Risk of ConcurrentModification?

Reply:
Actually looking more into this I think my change is wrong anyway. I reverted it to original version.